Add support for GeometryCollection fields

GJdan opened the pull request:
Fixes #126 
We have a project that has a need to support GeometryCollection fields.  After poking around I decided to try altering the GeometryField and GeometrySerializerMethodField to use GeoDjango's built-in geojson output instead of creating a GeoJSON object on the fly.  It seems to be a small change and fixes the issue in our project without breaking elsewhere.

nemesisdesign commented on rest_framework_gis/fields.py:
```diff
@@ -67,6 +62,18 @@ class GeoJsonDict(OrderedDict):
+                args = (geojson,)
+            except ValueError:
+                pass
+
```
could you remove this blank line please?

nemesisdesign commented on rest_framework_gis/fields.py:
```diff
@@ -67,6 +62,18 @@ class GeoJsonDict(OrderedDict):
     """
+    def __init__(self, *args, **kwargs):
+        """
+        If a string is passed attempt to pass it through json.loads, because it should be a geojson formatted string.
```
Could you please break the comment like this?
```
If a string is passed attempt to pass it through json.loads,
because it should be a geojson formatted string.
```

GJdan commented:
Sure.  I have to go to bed now, but I'll push the formatting fixes, and work on a test tomorrow.

GJdan commented:
@nemesisdesign I've added a test.  First, I apologize, I'm not yet as familiar with the testing system as I would like to be.  I ripped off the other geojson format tests above to keep some consistency, and also added assertJSONEqual because that's what made sense to me in this kind of test.

nemesisdesign commented on tests/django_restframework_gis_tests/tests.py:
```diff
@@ -584,3 +584,58 @@ def test_pickle(self):
+        else:
+            self.assertItemsEqual(json.dumps(response.data), json.dumps(expected))
+        self.assertContains(response, "Kool geometry collection geojson test")
+        self.assertJSONEqual(json.dumps(response.data), expected)
```
isn't this line redundant with lines 635-639? If yes I think you can keep this line and safely remove lines 635-639

GJdan replied:
Sorry, this was a misunderstanding on my part thinking that assertCountEqual and assertItemsEqual wouldn't do a thorough job because of their names, but on more research I see that they work very well.  Line 640 is there because it came with the tests I copied and pasted from test_geojson_format.  I have no problem removing it as well.

Removed line 641.

GJdan commented:
Done.  I think.  Unless you want me to remove the `self.assertContains` line?

nemesisdesign commented:
@GJdan no it's fine, I would just like to ask you one last thing: to squash the commits into one and put the commit message `Added support for GeometryCollection #126`, I could do this myself (squash and merge) but if you do it the history will look cleaner on github.

GJdan commented:
@nemesisdesign Done
